src/fs/root/tree.ts

```diff
@@ -39,7 +39,7 @@ import { PublicRootWasm } from "../v3/PublicRootWasm.js"
 type Dependents = {
   crypto: Crypto.Implementation
   depot: Depot.Implementation
-  manners: Manners.Implementation
+  manners: Manners.Implementation<any>
```

Comment:
I'm using `<any>` here, because I want to indicate that it doesn't matter what the type of the 'file system' is that the hooks are using, because we aren't using the hooks here. Another reason is if we were to use the `FileSystem` class, we'd be introducing some circular imports (same for all the other places I used `<any>`)

Reply:
Would like to avoid introducing `any` into the code base in all these places. Let's explore this to see if we can find an alternative.

First question is whether it is necessary to introduce the type parameter. It looks like the type parameter is referenced in the manners implementation.

https://github.com/fission-codes/webnative/blob/b381bea661eb826890d817bb3845dcfdf8fb6b00/src/components/manners/implementation.ts#L30-L31

Would it be possible to import the `FileSystem` as a type in this module instead?

```typescript
import type FileSystem from "../../fs/filesystem.js"
```

Other alternatives might be checking if we can use `unknown` instead of `any`, but removing the type parameter if we don't need it would be better.

Reply:
> Would like to avoid introducing `any` into the code base in all these places.

Me too.

> First question is whether it is necessary to introduce the type parameter

It is, otherwise we have circular imports (you can test this with `npm run test:imports`).
Of course, there might be another way around this issue, but I haven't found any yet.

> Other alternatives might be checking if we can use unknown instead of any

I have considered that, but that doesn't fit as well, because that means you'd need to be type checking in the implementation if you're actually dealing with a `FileSystem`. So that would be a weird experience for devs to write hooks.

`any` actually makes sense here, the type could be "anything" in this case, because the code doesn't use the hooks. Would be great if there was another way to indicate this to the type system, but I haven't found it.

Reply:
Ok, scratch that, I've implemented Philipp's suggestion that uses an interface for the file system.